docs/src/main/sphinx/optimizer/pushdown.rst

```diff
@@ -46,6 +46,22 @@ columns.
 If projection pushdown is succesful, the ``EXPLAIN`` plan for the query only
 accesses the relevant columns in the ``Layout`` of the ``TableScan`` operation.
 
+.. _dereference-pushdown:
+
+Dereference pushdown
+--------------------
+
+Projection pushdown and dereference pushdown limit access to relevant columns,
+except dereference pushdown is more selective. It limits access to only read the
+specified fields within a top level or nested ``ROW`` data type.
+
+For example, consider a table in the Hive connector that has a column
+formed from an array of rows, and the nested row has several fields. If a query
```

Comment:
can we just mention `row` instead of `array of rows`? I don't think pushdown of `a[1].x` is supported. Also, dereference operation after unnesting an array isn't optimized into a dereferencing lambda before unnesting. cc @electrum 

May be something like 

```
For example, consider a table in the Hive connector that has a ``ROW`` typed column. ....
```

Reply:
@phd3 thanks, I didn't realize that wasn't supported yet. We should update as you suggest